x-pack/plugins/lens/public/indexpattern_datasource/indexpattern_suggestions.test.tsx

```diff
@@ -1077,6 +1078,206 @@ describe('IndexPattern Data Source suggestions', () => {
       });
     });
   });
+  describe('#getDatasourceSuggestionsForVisualizeField', () => {
+    describe('with no layer', () => {
+      function stateWithoutLayer() {
+        return {
+          ...testInitialState(),
+          layers: {},
+        };
+      }
+
+      it('should return an empty array', () => {
+        const suggestions = getDatasourceSuggestionsForVisualizeField(
+          stateWithoutLayer(),
+          '1',
+          'source'
+        );
+
+        expect(suggestions).toEqual([]);
+      });
+    });
+    describe('with a previous empty layer', () => {
+      function stateWithEmptyLayer() {
+        const state = testInitialState();
+        return {
+          ...state,
+          layers: {
+            previousLayer: {
+              indexPatternId: '1',
+              columns: {},
+              columnOrder: [],
+            },
+          },
+        };
+      }
+
+      it('should return an empty array if the field does not exist', () => {
+        const suggestions = getDatasourceSuggestionsForVisualizeField(
+          stateWithEmptyLayer(),
+          '1',
+          'field_not_exist'
+        );
+
+        expect(suggestions).toEqual([]);
+      });
+
+      it('should apply a bucketed aggregation for a string field', () => {
+        const suggestions = getDatasourceSuggestionsForVisualizeField(
+          stateWithEmptyLayer(),
+          '1',
+          'source'
+        );
+
+        expect(suggestions).toContainEqual(
+          expect.objectContaining({
+            state: expect.objectContaining({
+              layers: {
+                previousLayer: expect.objectContaining({
+                  columnOrder: ['id1', 'id2'],
+                  columns: {
+                    id1: expect.objectContaining({
+                      operationType: 'terms',
+                      sourceField: 'source',
+                      params: expect.objectContaining({ size: 5 }),
+                    }),
+                    id2: expect.objectContaining({
+                      operationType: 'count',
+                    }),
+                  },
+                }),
+              },
+            }),
+            table: {
+              changeType: 'initial',
+              label: undefined,
+              isMultiRow: true,
+              columns: [
+                expect.objectContaining({
+                  columnId: 'id1',
+                }),
+                expect.objectContaining({
+                  columnId: 'id2',
+                }),
+              ],
+              layerId: 'previousLayer',
+            },
+          })
+        );
+      });
+
+      it('should apply a bucketed aggregation for a date field', () => {
```

Comment:
I'm not sure we need this test, it's testing logic that isn't part of the function itself